### DISCUSSION TITLE
An error occurred after installing the MRTK program in Unity.

### DISCUSSION THREAD

**Discussion by parsley777** [microsoft/MixedRealityToolkit-Unity#9528] [Q&A] [answered]
An error occurred after installing the MRTK program in Unity. 
![1](https://user-images.githubusercontent.com/80885893/111716917-5c4fb200-889a-11eb-885f-7f6af9eb1e34.jpg)
![2](https://user-images.githubusercontent.com/80885893/111716932-6376c000-889a-11eb-9ec3-dfe210e45d6f.jpg)
![3](https://user-images.githubusercontent.com/80885893/111716945-68d40a80-889a-11eb-91d0-4e48bc37eeb0.jpg)

1. I installed MRTK(Foundation, Example) in Unity.
 2. I went into the HandInteractionExamples Scene.
 3. I saw the bug, Console message is
'Unable to register IMixedRealityInputDeviceManager data provider (Windows Mixed Reality Device Manager) because the value of concreteType is null.
This may be caused by code being stripped during linking. The link.xml file in the MixedRealityToolkit.Generated folder is used to control code preservation.'

'Unable to register IMixedRealityInputDeviceManager data provider (Windows Mixed Reality Eye Gaze Provider) because the value of concreteType is null.
This may be caused by code being stripped during linking. The link.xml file in the MixedRealityToolkit.Generated folder is used to control code preservation.'

4. I took the route below to investigate the problem.
Hierarchy > MixedRealityToolkit > MixedRealityToolkit Script > Input > Input Data Providers > Windows Mixed Reality Device Manager > Type

5. The type item is marked in red, and there is an error.

6. There are two things marked in red ones.

Windows Mixed Reality Device Manager > Type
Windows Mixed Reality Eye Gaze Provider > Type

*My Unity, MRTK Version
 - Unity Version 2020.2.7
I tested Unity 2019.4.8 version, but the same problem occurred.
 - MRTK Version 2.6.1
I tested MRTK 2.5.3 version, but the same problem occurred.

*My Device
 - Hololens2

I need you help me.
I look forward to your reply.
Thank you.

**Comment by MaxWang-MS** [comment, marked as answer]
The issue is due to the fact that the default profile is based on the legacy XR pipeline, which is deprecated in Unity 2019 and removed in Unity 2020. The preferred (by Unity) XR pipeline in Unity 2019 and the only XR pipeline in Unity 2020 is the XR SDK pipeline.

The solution is to click on the MixedRealityToolkit GameObject in the hierarchy, look at the inspector window and then change the profile to `DefaultHololens2XRSDKConfigurationProfile` as you are targeting HoloLens 2 (if you are not you can use `DefaultXRSDKConfigurationProfile` and customize it as needed).

For more information on customizing profiles for XRSDK, please see [here](https://docs.microsoft.com/windows/mixed-reality/mrtk-unity/configuration/getting-started-with-mrtk-and-xrsdk) (profile related information is in the "Configuring MRTK for the XR SDK pipeline" section but it is highly recommended that you read the whole page, although you may skip the Oculus and OpenXR sections if you are not using them).

**Comment by parsley777** [comment]
Nice to meet you, Mr/Ms. MaxWang.
Thank you for your kind reply.
Your advise was very helpful to me.
Have a nice day.
Thank you.

Best regards,
Thomas.

2021년 3월 24일 (수) 오전 9:36, MaxWang-MS ***@***.***>님이 작성:
